src/KIM/kim_interactions.cpp

```diff
@@ -312,7 +315,6 @@ void KimInteractions::KIM_SET_TYPE_PARAMETERS(const std::string &input_line) con
           input->one(fmt::format("set type {} charge {}",ia+1,words[1]));
     }
   }
-  fclose(fp);
```

Comment:
@ellio167 This is the place, which causes issues on the C-Python interface when using `ase` & `lammpslib` and calling the `kim_interactions` multiple times in a loop.

Reply:
Thanks;  Yes, the second call to `fclose` is definitely a bug; both BSD (macOS) and Linux manpages indicate that calling `fclose` a second time gives undefined behavior.

Reply:
Like with pointers to memory location and malloc()/free(), the usual convention is to set a file pointer to NULL (or nullptr in C++11+) after it has been freed with fclose(). Then you can always add a check if the pointer is non-zero before attempting to close it. Of course that also requires the pointer to be initialized before use.